src/modules/commander/Commander.hpp

```diff
@@ -129,6 +138,17 @@ class Commander : public ModuleBase<Commander>, public ModuleParams
 	bool		_nav_test_passed{false};	/**< true if the post takeoff navigation test has passed */
 	bool		_nav_test_failed{false};	/**< true if the post takeoff navigation test has failed */
 
+	/* class variables used to check for airspeed sensor failure */
+	bool		_tas_check_fail{false};	/**< true when airspeed innovations have failed consistency checks */
+	hrt_abstime	_time_last_tas_pass{0};		/**< last time innovation checks passed */
+	hrt_abstime	_time_last_tas_fail{0};		/**< last time innovation checks failed */
+	const hrt_abstime TAS_INNOV_FAIL_DELAY = 1_s;	/**< time required for innovation levels to pass or fail (usec) */
+	bool		_tas_use_inhibit{false};	/**< true when the commander has instructed the control loops to not use airspeed data */
+	hrt_abstime	_time_tas_good_declared{0};	/**< time TAS use was started (uSec) */
+	hrt_abstime	_time_tas_bad_declared{0};	/**< time TAS use was stopped (uSec) */
+	hrt_abstime	_time_last_airspeed{0};		/**< time last airspeed measurement was received (uSec) */
+	char		*_airspeed_fault_type = new char[7];
```

Comment:
Re-reading your comment, I may have misunderstood it first time. Can you describe what you mean by using flags and why the objection to using strings?

Reply:
I was thinking of something like using more status flag booleans for each of the possibilities instead of strings.

Reply:
I know this uses some more memory with the 7 character class variable, however it avoids the necessity to duplicate branching wherever the message is printed. If you have an idea how this can be avoided, please provide an outline. TXS